shared/src/main/scala/VAny.scala

```diff
@@ -37,17 +38,40 @@ case class VFun(
   /** Call this function on the given arguments, using custom context variables.
     */
   def execute(
-      contextVarM: VAny,
-      contextVarN: VAny,
+      contextVarPrimary: VAny,
+      contextVarSecondary: VAny,
       args: Seq[VAny]
   )(using ctx: Context): VAny =
     Interpreter.executeFn(
       this,
-      Some(contextVarM),
-      Some(contextVarN),
+      Some(contextVarPrimary),
+      Some(contextVarSecondary),
       args = args
     )
 
+  def executeResult(
+      contextVarPrimary: VAny,
+      contextVarSecondary: VAny,
+      args: Seq[VAny]
+  )(using ctx: Context): VAny =
+    val res = Interpreter.executeFn(
+      this,
+      Some(contextVarPrimary),
+      Some(contextVarSecondary),
+      args = args
+    )
+
+    res match
+      case f: VFun =>
```

Comment:
Why execute the function if a function was returned? That seems dangerous

Reply:
Because when you're sorting a list or filtering a list by a function, you don't want to compare by a function. You want to compare by the result of calling that function.

Imagine you have something like

```
{0 < if negate end} :=abs
"some list of numbers" sort-lambda $abs then negate end
```

You don't want to order by the literal function `$abs`, you want to compare by the result.